### PR TITLE
Update to rxjs 6 and angular 6

### PR DESCRIPTION
**Pull request by NKjoep**
Hello! 

I hope this PR isn't unexpected and it might be considered as an input to perform the actual upgrade of the dependencies.

As you can see there aren't so many files to update. The tests are valid as well.

Please le me know if I can help further. 🍺

### CODE REVIEW

**Comment by alaszlotrend** [comment]
Consider changing the node version of [.travis.yml](https://github.com/Greentube/localize-router/blob/master/.travis.yml) (6 or higher)

**Comment by NKjoep** [comment]
@alaszlotrend thanks for letting me notice that! Since the LTS is version 8, I've updated with that.

**Comment by imsingh** [comment]
@NKjoep You probably should change the thing that I mentioned with `toPromise()`. It's method on Observable class and you don't really need to create Promise from scratch.

File: src/localize-router.parser.ts [hunk 1]
```diff
@@ -110,7 +106,12 @@ export abstract class LocalizeParser {
 
     /** translate routes */
     const res = this.translateRoutes(selectedLanguage);
-    return res.toPromise();
+    return new Promise((resolve, reject) => {
```
Comment (imsingh) [inline src/localize-router.parser.ts]: You don't need to create Promise here. Use `toPromise()` method on Observable.

**Comment by NKjoep** [comment]
@imsingh As far as I read, `toPromise()` was part of rxjs/operators version 5.5 and has been removed on v6 (see this thread https://github.com/ReactiveX/rxjs/issues/3482).

What do you think is the best option we have?

File: src/localize-router.parser.ts (hunk 1 above)
Comment (NKjoep) [reply src/localize-router.parser.ts]: @imsingh As far as I read, `toPromise()` was part of rxjs/operators version 5.5 and has been removed on v6 (see this thread https://github.com/ReactiveX/rxjs/issues/3482).

What do you think is the best option we have?

Comment (imsingh) [reply src/localize-router.parser.ts]: @NKjoep yeah it was removed as lettable operator. But it's now method on Observable class(same like previously). Take a look at https://github.com/ReactiveX/rxjs/issues/2868

**Comment by rafa-as** [comment]
some news here?

**Comment by venanciorodrigo** [comment]
I think we should merge this one, it's a big improvement ;)

**Comment by cappster** [comment]
When will this issue complete?

**Comment by PjGeeroms** [comment]
Can anyone tell me how to add this revision to my project instead of the old one? I'm working on a multilingual website and in need of this since it's running Angular 6. I downloaded this revision but where do i go from there?

**Comment by elmarbeckmann** [comment]
Great PR, does anybody have contact with guys like @manzonif or @meeroslav? Who can check/merge this request? @NKjoep were you in touch with them?

**Comment by NKjoep** [comment]
@elmarbeckmann No, I wasn't. I spontaneously opened this PR then asked to merge it.

**Comment by Roelensdam** [comment]
@NKjoep I was able to fetch your PR from the repo here but i can't make the build work... How did you build it ?

**Comment by NKjoep** [comment]
@Roelensdam I used the npm run build:

```log
18:31 $ rm -rf node_modules && npm run build

> localize-router@1.0.0-rc.4 build /Users/nkjoep/workspace/github/localize-router
> webpack

[at-loader] Using typescript@2.7.2 from typescript and "tsconfig.json" from /Users/nkjoep/workspace/github/localize-router/tsconfig.json.

[at-loader] Checking started in a separate process...

[at-loader] Ok, 0.517 sec.
Hash: 1817370beb4faaf793c4
Version: webpack 2.6.0
Time: 3325ms
                     Asset    Size  Chunks                    Chunk Names
    localize-router.umd.js  267 kB       0  [emitted]  [big]  main
localize-router.umd.js.map  265 kB       0  [emitted]         main
  [18] ./src/localize-router.config.ts 4.25 kB {0} [built]
  [19] ./src/localize-router.parser.ts 15.6 kB {0} [built]
  [20] ./~/rxjs/_esm5/internal/config.js 859 bytes {0} [built]
  [21] ./~/rxjs/_esm5/internal/util/ObjectUnsubscribedError.js 599 bytes {0} [built]
  [23] ./src/localize-router.service.ts 8.35 kB {0} [built]
  [24] ./~/rxjs/_esm5/internal/AsyncSubject.js 1.63 kB {0} [built]
  [25] ./~/rxjs/_esm5/internal/observable/of.js 808 bytes {0} [built]
  [32] ./src/localize-router-config-loader.ts 3.82 kB {0} [built]
  [33] ./src/localize-router.pipe.ts 2.99 kB {0} [built]
  [34] ./~/rxjs/_esm5/index.js 3.28 kB {0} [built]
  [58] ./src/localize-router.module.ts 6.13 kB {0} [built]
  [59] ./index.ts 497 bytes {0} [built]
  [60] ./src/util.ts 1.68 kB {0} [built]
  [63] ./~/rxjs/_esm5/internal/ReplaySubject.js 4.35 kB {0} [built]
  [64] ./~/rxjs/_esm5/internal/observable/ConnectableObservable.js 5.5 kB {0} [built]
    + 97 hidden modules
```

**Comment by ponkys** [comment]
@Roelensdam  I could build this PR using:
```
1. git fetch upstream pull/131/head:local-pr-131
2. rm -rf node_modules && npm i && npm run build
```

It would be great if we land this pr. Let me know if I can help with something else.

**Comment by Empereol** [comment]
@NKjoep , I've built this PR and patched it into my project. I don't get any `rxjs` errors but the routing is not working as expected. 

When changing language, the route is not changed/updated. And any localized routes also do not function.

I'm curious if this is the case for you or if this is a problem unique to my setup.

**Comment by elmarbeckmann** [comment]
@meeroslav Thank you so much for merging and fixing!!!

**Comment by meeroslav** [comment]
@elmarbeckmann you are welcome. Sorry for waiting so long for this

**Pull request merged**